### ISSUE TITLE
Not works on Arch

### ISSUE THREAD

**Issue by Dentrax** [b4b4r07/emoji-cli#20]
Hey,

Thanks for awesome `.zsh` plug-in! But unfortunately, it didn't work on my _Arch_. It shows up `fwd-i-search:` when i press <kbd>CTRL+S</kbd>. I installed it via `antigen`.

**Terminal:** _Alacritty_

Thanks

**Comment by b4b4r07** [comment]
I dont have Arch env.. So cannot reproduce 🤷‍♂ Someone can show me what happened in that time?

**Comment by cbrghton** [comment]
[Video](https://drive.google.com/open?id=146QkwOdNL5AcaJLQ82Pi1agwnusYoqRu)

it's more than anything to change the emoji-cli keybind, not for Arch

:v:

**Comment by WhyNotHugo** [comment]
Also Arch user here. Pressing ctrl+s just freezes the terminal for me -- nothing happens, but all keys stop responding.

Any hints on how to debug, or what information could be of use here? I'm happy to help debug this.

**Comment by delucca** [comment]
No updates on this? Same issue here...

**Comment by DonBras** [comment]
just put `stty -ixon` at the end of your ´.zshrc´. the freezing issue has nothing to do with emoji-cli :)

[https://unix.stackexchange.com/questions/72086/ctrl-s-hangs-the-terminal-emulator](https://unix.stackexchange.com/questions/72086/ctrl-s-hangs-the-terminal-emulator)

**Comment by WhyNotHugo** [comment]
Nothing like a solution to a problem accompanied with a history lesson. Works like a charm, thanks!

**Comment by WhyNotHugo** [comment]
It's not working _entirely_ right for me. It doesn't insert the emoji, but the _name_ of the emoji (the whole point is to get the emoji character into the terminal):

![recording](https://user-images.githubusercontent.com/730811/85167843-ffff2980-b258-11ea-9097-6a9457ff4ce2.gif)

**Comment by delucca** [comment]
Actually, my problem is not related to it hanging the terminal... When I do so, instead of opening the `emoji-cli` it shows the `fwd-i-search`, even with @DonBras proposed solution on `.zshrc`

```
~
❯
fwd-i-search: _
```

**Comment by WhyNotHugo** [comment]
@odelucca Sounds like you have C-s mapped to something else. Can you try with a clean `.zshrc`, and progressively add your current settings into and, to see which one triggers the breakage?

**Comment by DonBras** [comment]
@WhyNotHugo true... didn't try out to actually select an emoji :facepalm: 

@odelucca i had the same issue, but if you `source` the `emoji-cli.zsh` file at the end of your `.zshrc` it should work.. at least the emoji search part... selection is another thing.

**Comment by WhyNotHugo** [comment]
Hint: You also need to set `EMOJI_CLI_USE_EMOJI` to anything non-null for it to work. Otherwise it just prints the name and not an the selected emoji.

**Comment by DonBras** [comment]
@WhyNotHugo Sometimes the answer is written in the Readme :smile: 

So i guess we're done here now :tada:

**Comment by delucca** [comment]
Sorry for the late reply, but it still won't work. I've installed it with `zinit` (formerly `zplug`) and it is installed, as you can see here:

```txt
~
❯ zinit list
...
b4b4r07/emoji-cli *
...
```

Even if I change the `EMOJI_CLI_KEYBIND` to other key (`e`, for example) I can't manage to open the emoji dialog.

**Comment by WhyNotHugo** [comment]
@odelucca Have you what I mentioned [in the above comment](https://github.com/b4b4r07/emoji-cli/issues/20#issuecomment-646802413)?

Here's how I'd do it:

1. Comment out all our your `.zshrc`.
2. Open a terminal. Does it work?
3. If it does, uncomment a single line. Repeat until it stops working.
4. Once it stops working, you know exactly what line makes it fail.

If it doesn't work on step (2), then report back, there's something weirder broken.  
If you reach (4) and it's not obvious why it doesn't work, come back with the offending line.

**Comment by DonBras** [comment]
@odelucca I'm not familiar with zinit but maybe you have to put emoji-cli at the end of your plug-in list.

**Comment by delucca** [comment]
Hey guys,

Sorry for the late reply. I've removed **everything** from my `.zshrc` file, and left only the following:

```text
source ${HOME}/.zinit/bin/zinit.zsh
stty -ixon
zinit light "b4b4r07/emoji-cli"
```

But it is still not working... So, this isn't related to my `.zshrc` file.

**Comment by delucca** [comment]
no updates on this?

**Comment by delucca** [comment]
For reference, my test `.zshrc` file is:

```
source ${HOME}/.zinit/bin/zinit.zsh

autoload -U compinit

zinit snippet "OMZ::plugins/fzf/fzf.plugin.zsh"
zinit light "b4b4r07/emoji-cli"
```

and it doesn't work. When I tap `ctrl+s` nothing happens

**Comment by kkh913** [comment]
I had the same problem and came to visit this place. I solved the problem on my own and I would like to share how.

> TL;DR
> For users to use the plugin manager, e.g. `zplug`, the init script `~/.fzf.zsh` must be called before `zplug load`.

I'm pretty sure most of the people who have visited here have installed `fzf`. And `.zshrc`  has the following script:

```
[ -f ~/.fzf.zsh ] && source ~/.fzf.zsh
```

`emoji-cli` depends on `fzf`. So the script must be called "before" this plugin is loaded.

Here is my `.zshrc`

```
# .zshrc (use zplug)

source ~/.zplug/init.zsh


# Plugins
zplug "plugins/git",   from:oh-my-zsh
zplug "lib/completion",   from:oh-my-zsh
zplug 'lib/key-bindings', from:oh-my-zsh
zplug "lib/directories",  from:oh-my-zsh

zplug "zsh-users/zsh-syntax-highlighting"
zplug "zsh-users/zsh-autosuggestions"
zplug "denysdovhan/spaceship-prompt", use:spaceship.zsh, from:github, as:theme
zplug "b4b4r07/emoji-cli"

# zplug 'dracula/zsh', as:theme

# Install plugins if there are plugins that have not been installed
if ! zplug check --verbose; then
    printf "Install? [y/N]: "
    if read -q; then
        echo; zplug install
    fi
fi

[ -f ~/.fzf.zsh ] && source ~/.fzf.zsh 
export FZF_DEFAULT_OPTS="--layout=reverse --inline-info" # this option is just for me... 

# Then, source plugins and add commands to $PATH
zplug load # <-- please check this location
```

**Comment by HaleTom** [comment]
I changed the keybinding as ^s is a standard terminal pause (^q to resume).

```
EMOJI_CLI_KEYBIND='^o'
```

**Comment by babarot** [comment]
I’m sorry I cannot support this for now..
